### PR TITLE
Setup publishing with sbt-ci-release

### PR DESCRIPTION
**Pull request by joroKr21**
So that any maintainer can publish by pushing a tag.
Also fix semiauto docs and delete 2.11 specific code.

@kailuowang you would have to follow the steps here if we want to do this: https://github.com/olafurpg/sbt-ci-release

### CODE REVIEW

File: build.sbt [hunk 1]
```diff
@@ -93,54 +86,16 @@ lazy val crossVersionSharedSources: Seq[Setting[_]] =
...
-    tagRelease,
-    publishArtifacts,
-    setNextVersion,
-    commitNextVersion,
-    ReleaseStep(action = Command.process("sonatypeReleaseAll", _)),
-    pushChanges
+  scmInfo := Some(ScmInfo(url("https://github.com/typelevel/kittens"), "scm:git:git@github.com:typelevel/kittens.git")),
+  developers := List(
+    Developer("milessabin", "Miles Sabin", "", url("http://milessabin.com/blog")),
+    Developer("joroKr21", "Georgi Krastev", "joro.kr.21@gmail.com", url("https://twitter.com/Joro_Kr"))
```
Comment (joroKr21) [inline build.sbt:96]: @kailuowang if you want I will add you here as well

Comment (kailuowang) [reply build.sbt:96]: Thanks!

Comment (joroKr21) [reply build.sbt:97]: Oh no it's been almost a month 🙈 let me know if that works for you. And then after you setup https://github.com/olafurpg/sbt-ci-release we can merge this and should be able to release via pushing tags 🥳

Comment (kailuowang) [reply build.sbt:97]: @joroKr21 sorry I am still swamped. I just made you the admin of this repo. Would that enable you to setup the secrets?

**Review by kailuowang — APPROVED** [review]

File: build.sbt (hunk 1 above)
Comment (joroKr21) [reply build.sbt:97]: > sorry I am still swamped.

@kailuowang me as well 😅 

> I just made you the admin of this repo.

I do not know. How do you usually release?

**Pull request closed without merging**